narwhals/_pandas_like/dataframe.py

```diff
@@ -233,12 +236,28 @@ def select(
         )
         return self._from_native_frame(df)
 
-    def drop_nulls(self, subset: str | list[str] | None) -> Self:
+    def drop_nulls(
+        self: Self, subset: str | Selector | list[Selector | str] | None
+    ) -> Self:
         if subset is None:
             return self._from_native_frame(self._native_frame.dropna(axis=0))
-        subset = [subset] if isinstance(subset, str) else subset
+
+        from narwhals.selectors import Selector
+
         plx = self.__narwhals_namespace__()
-        return self.filter(~plx.any_horizontal(plx.col(*subset).is_null()))
+        subset_: Iterable[str | Selector] = (
+            [subset] if isinstance(subset, (str, Selector)) else subset
+        )
+
+        column_names = [c for c in subset_ if isinstance(c, str)]
+        selectors = [c.is_null()._call(plx) for c in subset_ if isinstance(c, Selector)]
```

Comment:
at the moment non-string columns are supported here
```python
In [18]: @nw.narwhalify
    ...: def func(df):
    ...:     return df.drop_nulls(subset=[0])
    ...:

In [19]: func(pd.DataFrame({0: [1,2,None]}))
Out[19]:
     0
0  1.0
1  2.0
```
so if I'm reading this right we'd risk regressing

Reply:
I think I only maintained the support for dask, but that would be an easy fix as in `_dask/dataframe.py` line 154